server/src/main/java/org/elasticsearch/plugins/internal/rewriter/InternalQueryRewriter.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.plugins.internal.rewriter;
+
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryRewriteContext;
+
+/**
+ * Enables modules and plugins to intercept and rewrite queries during the query rewrite phase on the coordinator node.
+ */
+public interface InternalQueryRewriter {
```

Comment:
Why is this needed? It doesn't seem to be used outside of the `QueryRewriteInterceptor`.

Reply:
You're right, that's cleanup from an initial approach I took during the refactor. Fixed in 068c4bbfd49da90a3a150462d67f34691fc10da6